src/carp.cc

```diff
@@ -73,24 +58,16 @@ carpInit(void)
         if (p->weight == 0)
             continue;
 
-        ++n_carp_peers;
-
         W += p->weight;
+
+        CarpPeers.emplace_back(p);
```

Comment:
Please move this emplace_back() line to where we used to increment `n_carp_peers`. We want to emphasize that those two `continue` conditions (plus a new condition I am requesting in another change request, plus any future conditions), are the conditions that determine CarpPeers membership. Adding any other code (e.g., weight increment) between the conditions and emplace_back() dilutes that message and increases the probability of future bugs, where, for example, weight is incremented, but the corresponding cache_peer is not added.